File: app/src/main/java/com/elementary/tasks/core/services/PermanentBirthdayReceiver.java
package com.elementary.tasks.core.services;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.elementary.tasks.core.utils.Notifier;
import com.elementary.tasks.core.utils.Prefs;

/**
 * Copyright 2017 Nazar Suhovich
 * <p/>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p/>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p/>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
public class PermanentBirthdayReceiver extends BroadcastReceiver {

    public static final int BIRTHDAY_PERM_ID = 356665;
    public static final String ACTION_SHOW = "com.elementary.tasks.birthday.SHOW";
    public static final String ACTION_HIDE = "com.elementary.tasks.birthday.HIDE";

    @Override
    public void onReceive(Context context, Intent intent) {
        if (!Prefs.getInstance(context).isBirthdayPermanentEnabled()) {
            Notifier.hideNotification(context, BIRTHDAY_PERM_ID);
            return;
        }
        if (intent != null) {
            String action = intent.getAction();
            if (action != null && action.matches(ACTION_SHOW)) {
                Notifier.showBirthdayPermanent(context);
            } else {
                Notifier.hideNotification(context, BIRTHDAY_PERM_ID);
            }
        } else {
            Notifier.hideNotification(context, BIRTHDAY_PERM_ID);
        }
    }
}

File: app/src/main/java/com/elementary/tasks/core/services/PermanentBirthdayReceiver.kt
package com.elementary.tasks.core.services

import android.content.Context
import android.content.Intent
import com.elementary.tasks.Actions

class PermanentBirthdayReceiver : BaseBroadcast() {

  override fun onReceive(context: Context, intent: Intent?) {
    if (!prefs.isBirthdayPermanentEnabled) {
      notifier.cancel(BIRTHDAY_PERM_ID)
      return
    }
    if (intent != null) {
      val action = intent.action
      if (action != null && action.matches(ACTION_SHOW.toRegex())) {
        notifier.showBirthdayPermanent()
      } else {
        notifier.cancel(BIRTHDAY_PERM_ID)
      }
    } else {
      notifier.cancel(BIRTHDAY_PERM_ID)
    }
  }

  companion object {

    const val BIRTHDAY_PERM_ID = 356665
    const val ACTION_SHOW = Actions.Birthday.ACTION_SB_SHOW
    const val ACTION_HIDE = Actions.Birthday.ACTION_SB_HIDE
  }
}
